### DISCUSSION TITLE
Any insights into seemingly random test failures appearing with new macOS 12 image

### DISCUSSION THREAD

**Discussion by DanPuzzuoli** [actions/virtual-environments#8539] [General questions]
Hi,

Recently some macOS 12 CI tests in a package I maintain have started failing randomly (e.g. [here](https://github.com/Qiskit-Extensions/qiskit-dynamics/actions/runs/6497046734/job/17645301742?pr=263)). The especially weird aspect of this is that these failures are random (sometimes they pass, sometimes they fail), but the code in the tests is fully deterministic (as far as I am aware).

I'm at a bit of a loss - no idea what could be causing this, so thought I'd ask here to see if anyone has any insights.

**Comment by v-sergei-pyshnoi** [comment]
Hi @DanPuzzuoli . Can you please check image version in "Set up job" step for failed add passed tests ? Is there was the same or different?

**Comment by trzopekm** [comment]
@sergei-pyshnoi we experience random failures as well but in our case it's during yarn package installation using ubuntu-latest images (both 22.04 and 20.04). It started happening after Ubuntu 22.04 image update from 20230924.1.0 to 20231001.1.0, can you please help with that issue?
